src/App.tsx

```diff
@@ -38,58 +38,110 @@ export function getReorderedGoods(
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
+  switch (sortType) {
+    case SortType.ALPHABET: visibleGoods.sort(
+      (a, b) => a.localeCompare(b),
+    ); break;
+
+    case SortType.LENGTH: visibleGoods.sort(
+      (a, b) => a.length - b.length,
+    ); break;
+
+    default: break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   return visibleGoods;
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleAlphabeticSort = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  handleLengthSort = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  handleReverseSort = () => {
+    this.setState((state) => ({ isReversed: !state.isReversed }));
+  };
+
+  handleReset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const visibleGoods = getReorderedGoods(goodsFromServer, { ...this.state });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={`button is-info ${sortType !== SortType.ALPHABET
+              ? 'is-light'
+              : ''}`}
+            onClick={this.handleAlphabeticSort}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={`button is-success ${sortType !== SortType.LENGTH
+              ? 'is-light'
+              : ''}`}
```

Comment:
suggestion: I'd rather use [classnames](https://www.npmjs.com/package/classnames) library here (I use it often when writing real commercial stuff)